File: src/backend/Source/Permissions/Allow.cs
namespace Backend.Permissions;

// ReSharper disable InconsistentNaming

/// <summary>
/// Centralized catalog of permission name constants used to authorize
/// endpoint and service operations across the backend.
/// </summary>
public partial class Allow
{
    public const string User_View = "User.View";
    public const string User_Create = "User.Create";
    public const string User_Update = "User.Update";
    public const string User_Delete = "User.Delete";

    public const string Role_View = "Role.View";
    public const string Role_Create = "Role.Create";
    public const string Role_Update = "Role.Update";
    public const string Role_Delete = "Role.Delete";
    public const string Role_ChangePermissions = "Role.ChangePermissions";

    public const string File_Delete = "File.Delete";

    public const string Tenant_View = "Tenant.View";
    public const string Tenant_Detail = "Tenant.Detail";
    public const string Tenant_Create = "Tenant.Create";
    public const string Tenant_Update = "Tenant.Update";
    public const string Tenant_Suspend = "Tenant.Suspend";
    public const string Tenant_Reactivate = "Tenant.Reactivate";
    public const string Tenant_Delete = "Tenant.Delete";

    public const string TenantMember_View = "TenantMember.View";
    public const string TenantMember_Add = "TenantMember.Add";
    public const string TenantMember_UpdateRoles = "TenantMember.UpdateRoles";
    public const string TenantMember_Remove = "TenantMember.Remove";

    public const string Edition_View = "Edition.View";
    public const string Edition_Create = "Edition.Create";
    public const string Edition_Update = "Edition.Update";
    public const string Edition_Delete = "Edition.Delete";

    public const string FeatureValue_View = "FeatureValue.View";
    public const string FeatureValue_Manage = "FeatureValue.Manage";
}

File: src/frontend/web/allow.ts
/**
 * Central registry of permission keys used throughout the app to gate access to user, role, file and tenancy features.
 * Every entry mirrors a constant of the same name in the API's `Allow` catalog.
 */
export const Allow = {
  User_View: 'User.View',
  User_Create: 'User.Create',
  User_Update: 'User.Update',
  User_Delete: 'User.Delete',

  Role_View: 'Role.View',
  Role_Create: 'Role.Create',
  Role_Update: 'Role.Update',
  Role_Delete: 'Role.Delete',
  Role_ChangePermissions: 'Role.ChangePermissions',

  File_Delete: 'File.Delete',

  Tenant_View: 'Tenant.View',
  Tenant_Detail: 'Tenant.Detail',
  Tenant_Create: 'Tenant.Create',
  Tenant_Update: 'Tenant.Update',
  Tenant_Suspend: 'Tenant.Suspend',
  Tenant_Reactivate: 'Tenant.Reactivate',
  Tenant_Delete: 'Tenant.Delete',

  TenantMember_View: 'TenantMember.View',
  TenantMember_Add: 'TenantMember.Add',
  TenantMember_UpdateRoles: 'TenantMember.UpdateRoles',
  TenantMember_Remove: 'TenantMember.Remove',

  Platform_Administration: 'Platform.Administration',
} as const
